### ISSUE TITLE
NPE/BlockPlaceEvent errors

### ISSUE THREAD

**Issue by SlimeDog** [flori-schwa/VarLightOld#24]
Spigot 1.15.2 (Jenkins 2622)
LightAPI 3.4.2 ([bottomdumber fork](https://www.spigotmc.org/resources/lightapi-fork.48247/history))
VarLight 1.8.2-MC-1.15.1 (there is not 1.15.2 version)

We are seeing quite a few of these as we move around. There is only one VarLight configured, in a different world.

```
[17:05:24] [Server thread/ERROR]: Could not pass event BlockPlaceEvent to VarLight v1.8.2
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:541) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:528) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at com.rngservers.irondoors.events.Events.onIronDoor(Events.java:39) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor289.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:541) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:528) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:442) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.PlayerInteractManager.a(PlayerInteractManager.java:441) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1250) ~[?:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:27) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeAll(SourceFile:103) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:894) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.NullPointerException
        at me.shawlaf.varlight.spigot.VarLightPlugin.onBlockPlace(VarLightPlugin.java:435) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor469.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar-2020-02-02-2025:git-Spigot-c574e08-807a677]
        ... 27 more
```

**Comment by flori-schwa** [comment]
This is somehow caused by placing a block while holding nothing in your hand. Something I thought was impossible, but you seem to have a plug-in with such a functionality installed (com.rngservers.irondoors). Will have to add a check for that

**Comment by flori-schwa** [comment]
What's even more confusing is that [org.bukkit.event.block.BlockPlaceEvent#getItemInHand](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/block/BlockPlaceEvent.html#getItemInHand--) is marked [NotNull](https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html#notnull), meaning `com.rngservers.irondoors` is violating that rule

**Comment by flori-schwa** [comment]
Fixed in [1.8.3](https://github.com/flori-schwa/VarLight/releases/tag/1.8.3)
